internal/events/aggregator.go

```diff
@@ -532,6 +532,18 @@ func (ag *aggregator) processMessage(ctx context.Context, manifest *core.BatchMa
 	return nil
 }
 
+func needsTokenTransfer(msg *core.Message) bool {
+	return (msg.TxParent != nil && msg.TxParent.Type == core.TransactionTypeTokenTransfer) ||
+		msg.Header.Type == core.MessageTypeTransferBroadcast ||
```

Comment:
These "special" message types can be deprecated if we go this route, as `TxParent.Type` would be the new/authoritative way to identify an associated transfer/approval.